src/datasets/formatting/formatting.py

```diff
@@ -194,11 +194,11 @@ def _arrow_array_to_numpy(self, pa_array: pa.Array) -> np.ndarray:
                 array: List = pa_array.to_numpy(zero_copy_only=zero_copy_only).tolist()
         if len(array) > 0:
             if any(
-                (isinstance(x, np.ndarray) and (x.dtype == np.object or x.shape != array[0].shape))
+                (isinstance(x, np.ndarray) and (x.dtype == np.dtype("object") or x.shape != array[0].shape))
```

Comment:
The warning message suggests to use `object` instead.
```suggestion
                (isinstance(x, np.ndarray) and (x.dtype == object or x.shape != array[0].shape))
```